src/client/components/ConsentsSubHeader.tsx

```diff
@@ -40,224 +48,177 @@ const h1ResponsiveText = css`
+    }
 
-const pageProgression = css`
-  margin-top: ${space[5]}px;
-  margin-bottom: 0;
-  li {
-    color: ${neutral[60]};
-    &.active,
-    &.complete {
-      color: ${neutral[0]};
+    &:first-of-type > span {
+      left: -${CIRCLE_RADIUS}px;
+      text-align: left;
+
+      /**
+       * This allows us to force only "Create password" to wrap on mobile
+       */
+      ${until.mobileLandscape} {
+        white-space: normal;
+        width: 60px;
+      }
     }
-    &::after {
-      background-color: ${neutral[60]};
+
+    &:last-of-type > span {
+      right: -${CIRCLE_RADIUS}px;
+      left: auto;
+      text-align: right;
     }
+
     &::before {
-      border: 2px solid ${neutral[60]};
-      background-color: white;
-    }
-    &:first-child::before {
-      left: 0;
-    }
-    &:last-child::before {
-      background-color: ${neutral[60]};
+      content: '';
+      border-radius: 50%;
+      width: ${CIRCLE_DIAMETER}px;
+      height: ${CIRCLE_DIAMETER}px;
+      position: absolute;
+      left: -${CIRCLE_RADIUS}px;
+      /* TODO: apply this to psuedo elements globally */
+      box-sizing: border-box;
+      ${status === 'active' &&
+      `
+        background: ${neutral[100]};
+        border: ${COMPLETED_BORDER_SIZE}px solid ${COMPLETED_COLOR};
+      `}
+      ${status === 'pending' &&
+      `
+        background: ${neutral[100]};
+        border: ${PENDING_BORDER_SIZE}px solid ${PENDING_COLOR};
+      `}
+      ${status === 'complete' &&
+      `
+        background: ${brand[400]}
+      `}
     }
-    &:last-child::after {
-      border: 2px solid ${neutral[60]};
-      background-color: white;
-    }
-  }
-`;
 
-const ul = css`
-  display: flex;
-  list-style: none;
-  justify-content: space-between;
-  height: 54px;
-  padding: 0;
-  margin: 0;
+    ${status === 'active' && textSans.xxsmall({ fontWeight: 'bold' })}
+  `;
+};
+
+const progressWrapper = css`
+  padding: 0 ${CIRCLE_RADIUS}px;
+  overflow: hidden;
 `;
 
-const maxContentOverride = (isLastPage: boolean) => css`
-  width: ${isLastPage ? 'max-content' : 'initial'};
+const hiddenText = css`
+  display: none;
 `;
 
-const PageProgression = ({
-  pages,
-  current,
-  cssOverrides,
-}: {
-  pages: string[];
-  current?: string;
-  cssOverrides?: SerializedStyles | SerializedStyles[];
-}) => {
-  const active = current ? pages.indexOf(current) : 0;
-  const getClassName = (i: number) => {
-    switch (true) {
-      case i === active:
-        return 'active';
-      case i < active:
-        return 'complete';
-      default:
-        return '';
-    }
-  };
+export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
+  const active = current
+    ? (CONSENTS_PAGES_ARR as string[]).indexOf(current)
+    : 0;
+
+  const pageProgression = current && (
+    <div css={[progressWrapper, autoRow()]}>
+      <ol css={ol(active)}>
+        {CONSENTS_PAGES_ARR.map((page, index) => {
+          const status: PageStatus = (() => {
```

Comment:
thoughts on splitting out this unnamed function into a named one?

Reply:
```diff
diff --git a/src/client/components/ConsentsSubHeader.tsx b/src/client/components/ConsentsSubHeader.tsx
index 9f882d95..8909b6be 100644
--- a/src/client/components/ConsentsSubHeader.tsx
+++ b/src/client/components/ConsentsSubHeader.tsx
@@ -177,6 +177,16 @@ const screenReaderTextStyles = css`
   ${visuallyHidden}
 `;
 
+const getPageStatus = (index: number, active: number): PageStatus => {
+  if (index === active) {
+    return 'active';
+  }
+  if (index < active) {
+    return 'complete';
+  }
+  return 'pending';
+};
+
 export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
   const active = current
     ? (CONSENTS_PAGES_ARR as string[]).indexOf(current)
@@ -186,15 +196,7 @@ export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
     <div css={[progressWrapper, autoRow()]}>
       <ol css={ol(active)}>
         {CONSENTS_PAGES_ARR.map((page, index) => {
-          const status: PageStatus = (() => {
-            if (index === active) {
-              return 'active';
-            }
-            if (index < active) {
-              return 'complete';
-            }
-            return 'pending';
-          })();
+          const status = getPageStatus(index, active);
           const screenReaderText = {
             active: 'Current: ',
             complete: 'Completed: ',
```

Something like this? I like IIFEs sometimes when I need to assign the result of something slightly complex, that can't just be a single line expression, but also isn't really used anywhere else so assigning a name to the function seems a bit redundant.

Nested ternaries could be a solution for this sort of thing, if Prettier formatted them in a way where they could be digested easily at a glance. But sadly that is not the case.

Reply:
yeah it's a tricky one as there are good arguments for both sides, having it as an iife helps when I'm reading code sometimes so I'm not constantly scrolling around. i'm cool with keeping it like it is here